### PR TITLE
feat(MaticJS): adds zkevm client

### PR DESCRIPTION
**Pull request by ashucoder9**

### CODE REVIEW

File: docs/develop/ethereum-polygon/matic-js/zkevm/common-methods.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+## isWithdrawExitable
+
+This method checks if the withdrawal process can be exited.
+
+```js
+const isWithdrawExitable = await hermezClient.isWithdrawExitable(tx hash);
+```
+
+## isExited
+
```
Comment (nitinmittal23) [inline docs/develop/ethereum-polygon/matic-js/zkevm/common-methods.md:39]: here definition is missed

File: docs/develop/ethereum-polygon/matic-js/zkevm/erc20.md [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+```js
+// get balance of user
+const balance = await erc20Token.getBalance(<user Address>);
+```
+
+## Approve Methods
+
+### approve
+
+`approve` method can be used to approve required amount on the root token. It is required in order to deposit amount on the zkEVM network.
```
Comment (nitinmittal23) [inline docs/develop/ethereum-polygon/matic-js/zkevm/erc20.md:59]: this is for both root and child token

File: docs/develop/ethereum-polygon/matic-js/zkevm/erc20.md [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+```js
+// approve 1000 amount
+const result = await erc20Token.approve(1000, {
+    spenderAddress: <spender address value>
+});
+```
+
+### approveMax
+
+`approveMax` method can be used to approve maximum amount on the root token.
```
Comment (nitinmittal23) [inline docs/develop/ethereum-polygon/matic-js/zkevm/erc20.md:86]: this is also for both tokens, root and child

Comment (nitinmittal23) [reply docs/develop/ethereum-polygon/matic-js/zkevm/erc20.md:86]: this is same everywhere in approval, allowance, isApprovalNeeded

File: docs/develop/ethereum-polygon/matic-js/zkevm/common-methods.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  - hermez client
+image: https://wiki.polygon.technology/img/polygon-logo.png
+---
+
+## isDeposited
+
+The `isDeposited` method can be used to check if a deposit has been completed.
+
+```js
+const isDeposited = await hermezClient.isDeposited(tx hash);
```
Comment (ashucoder9) [inline docs/develop/ethereum-polygon/matic-js/zkevm/common-methods.md:19]:
```suggestion
const isDeposited = await hermezClient.isDeposited(tx_hash);
```

File: docs/develop/ethereum-polygon/matic-js/zkevm/common-methods.md (hunk 1 above)
Comment (ashucoder9) [inline docs/develop/ethereum-polygon/matic-js/zkevm/common-methods.md:39]:
```suggestion

isExited method checks if a withdrawal has been exited. It returns a boolean value.

```

File: docs/develop/ethereum-polygon/matic-js/zkevm/erc20.md (hunk 2 above)
Comment (ashucoder9) [inline docs/develop/ethereum-polygon/matic-js/zkevm/erc20.md:59]:
```suggestion
The `approve` method can approve the required amount on the root and child token. It is needed in order to deposit the amount on the zkEVM network.
```

File: docs/develop/ethereum-polygon/matic-js/zkevm/erc20.md (hunk 3 above)
Comment (ashucoder9) [inline docs/develop/ethereum-polygon/matic-js/zkevm/erc20.md:86]:
```suggestion
The `approveMax` method can approve the maximum amount on the root and child tokens.
```

File: docs/develop/ethereum-polygon/matic-js/zkevm/erc20.md [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+```js
+// approve 100 amount
+const result = await erc20Token.approveMax({
+    spenderAddress: <spender address value>
+});
+```
+
+### isApprovalNeeded
+
+`isApprovalNeeded` checks if an approval is needed for the root token.
```
Comment (ashucoder9) [inline docs/develop/ethereum-polygon/matic-js/zkevm/erc20.md:110]:
```suggestion
`isApprovalNeeded` checks if approval is needed for the root or child token.
```

File: docs/develop/ethereum-polygon/matic-js/pos/index.md [hunk 6]
```diff
@@ -6,9 +6,9 @@ keywords:
 description: 'POSClient allows you to interact with POS Bridge.'
 ---
 
-`maticjs` provides `POSClient` to interact with **POS** Bridge.
+MaticJS provides `POSClient` to interact with the **POS** Bridge.
```
Comment (ashucoder9) [inline docs/develop/ethereum-polygon/matic-js/pos/index.md:9]:
```suggestion
MaticJS provides `POSClient` to interact with the **PoS Bridge**.
```

**Review by rahuldamodar94 — APPROVED** [review]

**Pull request merged**